src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
     @Override
     public void add(T value) {
+        Node<T> newElement = new Node<>(null, value, null);
+        if (size == 0) {
+            head = newElement;
+        } else {
+            newElement.prev = tail;
+            tail.next = newElement;
+        }
+        tail = newElement;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < 0) {
+            throw new IndexOutOfBoundsException("Index: "
+                    + index + " is out of bounds this list size: " + size);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> nextElement = findElementByIndex(index);
+        Node<T> addedElement = new Node<>(nextElement.prev, value, nextElement);
+        if (nextElement.prev != null) {
+            nextElement.prev.next = addedElement;
+        } else {
+            head = addedElement;
+        }
+        nextElement.prev = addedElement;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkPositionIndex(index);
+        return findElementByIndex(index).value;
```

Comment:
In findElementByIndex() first line you call checkPositionIndex(). Do we need it here in this case?